php/libraries/User.class.inc

```diff
@@ -317,6 +317,21 @@ class User extends UserPermissions
     }
 
     /**
+     * Checks if the user is in at least one study site
+     *
+     * @return boolean
+     */
+    function isInOneStudySite()
+    {
+        $site_arr         = $this->userInfo['CenterIDs'];
+        foreach ($site_arr as $key=>$val) {
+            $site[$key]        = & Site::singleton($val);
+            $isStudySite[$key] = $site[$key]->isStudySite();
```

Comment:
can't you just have an early return here and then return false if nothing is found in the loop?